trino/auth.py

```diff
@@ -377,53 +381,53 @@ def _get_token(self, token_server, response, **kwargs):
 
         raise exceptions.TrinoAuthError("Exceeded max attempts while getting the token")
 
-    def _get_token_from_cache(self, host: str) -> Optional[str]:
+    def _get_token_from_cache(self, host: Optional[str]) -> Optional[str]:
         with self._token_lock:
             return self._token_cache.get_token_from_cache(host)
 
-    def _store_token_to_cache(self, host: str, token: str) -> None:
+    def _store_token_to_cache(self, host: Optional[str], token: str) -> None:
         with self._token_lock:
             self._token_cache.store_token_to_cache(host, token)
 
     @staticmethod
-    def _determine_host(url) -> Optional[str]:
+    def _determine_host(url: Optional[str]) -> Any:
```

Comment:
why is this returning `Any`?

Reply:
Expected return type here is `Union[str, None, bytes]` but if I change it then it complains about:
```
trino/auth.py:285: error: Argument 1 to "_get_token_from_cache" of "_OAuth2TokenBearer" has incompatible type "Union[str, None, bytes]"; expected "Optional[str]"  [arg-type]
trino/auth.py:343: error: Argument 1 to "_store_token_to_cache" of "_OAuth2TokenBearer" has incompatible type "Union[str, None, bytes]"; expected "Optional[str]"  [arg-type]
trino/auth.py:351: error: Argument 1 to "_get_token_from_cache" of "_OAuth2TokenBearer" has incompatible type "Union[str, None, bytes]"; expected "Optional[str]"  [arg-type]
```

It's probably not worth to annotate it with `Union[str, None, bytes]` since the return type is `str`.  Putting `Any` as to satisfy mypy.